partner_validate_vat/res_partner.py

```diff
@@ -24,47 +24,37 @@
 #
 #
 
-from openerp.osv import osv
-from openerp.tools.translate import _
+from openerp import models, api, _
+from openerp.exceptions import ValidationError
 
 
-class ResPartner(osv.Model):
+class ResPartner(models.Model):
     _inherit = 'res.partner'
 
-    def _check_vat_uniqueness(self, cr, uid, ids, context=None):
+    @api.multi
+    @api.constrains('vat')
```

Comment:
Really, that is not a problem because you could use a extra field computed with store=True just for parent_id is False and use it directly in the sql-constraint
Something like as https://github.com/Vauxoo/addons-vauxoo/pull/1205